Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -521,6 +519,26 @@ void ReflectometryReductionOne::exec() {
                          stitchingStart, stitchingDelta, stitchingEnd,
                          stitchingStartOverlap, stitchingEndOverlap);
 
+  OptionalMatrixWorkspace_sptr detectorEfficiencyCorrection;
+  MatrixWorkspace_sptr decTemp = getProperty("DetectorEfficiencyCorrection");
+  if (decTemp) {
+    if (decTemp->getAxis(0)->unit()->unitID() != "Wavelength") {
+      throw std::invalid_argument(
+          "Detector Efficiency Correction workspace x-units must be in "
+          "wavelength");
+    }
+    for (size_t i = 0; i < decTemp->getNumberHistograms(); i++) {
+      for (size_t j = 0; j < decTemp->blocksize(); j++) {
+        if (decTemp->y(i)[j] == 0) {
+          throw std::invalid_argument(
+              "The DetectorEfficiencyCorrection  "
+              "workspace cannot contain any zero values.");
+        }
+      }
+    }
```

Comment:
The right place to check that input properties are what we expect is the method `validateInputs()`. As an example see [here](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/MaxEnt.cpp#L168-L186)